pkg/iscsi/iscsi_util.go

```diff
@@ -470,23 +473,33 @@ func (util *ISCSIUtil) DetachDisk(
 
 	pathExists, pathErr := mount.PathExists(targetPath)
 	if pathErr == nil && !pathExists {
-		glog.Warningf(
+		logrus.Warningf(
 			"Warning: Unmount skipped because path does not exist: %v",
 			targetPath,
 		)
 		return nil
 	}
 
 	if err = c.mounter.Unmount(targetPath); err != nil {
-		glog.Errorf(
+		logrus.Errorf(
 			"iscsi detach disk: failed to unmount: %s\nError: %v",
 			targetPath, err,
 		)
 		return err
 	}
-	cnt--
-	if cnt != 0 {
-		return nil
+
+	// fetch the reference count again
+	_, cnt, err = mount.GetDeviceNameFromMount(c.mounter, targetPath)
```

Comment:
iSCSI stale sessions will still exist in a case where after `unmount` of disk if pod restarts.